### PR TITLE
Unroll dragons + MTCopier

### PR DESCRIPTION
**Pull request by Squadrick**

### CODE REVIEW

File: include/shadesmar/memory/dragons.h [hunk 1]
```diff
@@ -196,12 +205,158 @@ class AvxUnrollCopier : public Copier {
...
+    n = SHMALIGN(n, sizeof(typename BaseCopierT::PtrT)) /
+        sizeof(typename BaseCopierT::PtrT);
+    std::vector<std::thread> threads;
+    threads.reserve(nthreads);
+
+#ifdef __APPLE__
+    lldiv_t per_worker = div((int64_t)n, nthreads);
+#else
+    ldiv_t per_worker = div((int64_t)n, nthreads);
+#endif
```
Comment (Squadrick) [inline include/shadesmar/memory/dragons.h:320]: Gotta love this stupid difference.

Comment (shrijitsingh99) [reply include/shadesmar/memory/dragons.h:348]: Won't this be solvable if [std::lldiv_t](https://en.cppreference.com/w/cpp/numeric/math/div) is explicitly called?

Comment (Squadrick) [reply include/shadesmar/memory/dragons.h:348]: Brilliant; good catch.

Comment (Squadrick) [reply include/shadesmar/memory/dragons.h:348]: Better yet: `auto per_worker = ...`

Comment (shrijitsingh99) [reply include/shadesmar/memory/dragons.h]: Hahaha yeah, that's the best way.

**Pull request merged**